app/src/main/java/ch/sheremet/dasfoo/rover/android/client/MainActivity.java

```diff
@@ -107,14 +104,23 @@ private void enableButtons(final boolean isEnabled) {
         mReadEncodersButton.setEnabled(isEnabled);
     }
 
-    public class GrpcTask extends AsyncTask<AbstractGrpcTaskExecutor, Void, String> {
+    public class GrpcTask extends AsyncTask<AbstractGrpcTaskExecutor, Void, String>
+            implements ProviderInstaller.ProviderInstallListener {
+        private static final int PLAY_SERVICES_RESOLUTION_REQUEST = 9000;
+        private static final String PROVIDER_NOT_INSTALLED =
+                "The security provider isn't available";
```

Comment:
I think we should elaborate, e.g.: `The security provider installation failed, encrypted communication is not available`